dataprep/eda/distribution/render.py

```diff
@@ -1822,145 +1880,200 @@ def render_two_num(
         "layout": [panel.child for panel in tabs],
         "meta": [panel.title for panel in tabs],
         "container_width": plot_width + 80,
+        "how_to_guide": htgs,
     }
 
 
-def render_two_cat(
-    itmdt: Intermediate,
-    plot_width: int,
-    plot_height: int,
-) -> Dict[str, Any]:
+def render_two_cat(itmdt: Intermediate, cfg: Config) -> Dict[str, Any]:
     """
-    Render plots from plot(df, x, y) when x and y are categorical columns
+    Create visualizations for plot(df, Nominal, Nominal)
     """
+    # pylint: disable=too-many-locals
+    plot_width = cfg.plot.width if cfg.plot.width else 972
+    plot_height = cfg.plot.height if cfg.plot.height else 300
+
     tabs: List[Panel] = []
+    htgs: Dict[str, List[Tuple[str, str]]] = {}
     df = itmdt["data"].to_frame("cnt").reset_index()
     x, y = itmdt["x"], itmdt["y"]
-
-    # parse the dataframe to consist of the ngroups largest x groups
-    xgrps = df.groupby(x)["cnt"].sum()
-    x_lrgst = xgrps.nlargest(itmdt["ngroups"])
-    df = df[df[x].isin(x_lrgst.index)]
-    stats = {f"{x}_ttl": len(xgrps), f"{x}_shw": len(x_lrgst)}
-
-    # parse the dataframe to consist of the nsubgroups largest y groups
-    ygrps = df.groupby(y)["cnt"].sum()
-    y_lrgst = ygrps.nlargest(itmdt["nsubgroups"])
-    df = df[df[y].isin(y_lrgst.index)]
-    stats.update(zip((f"{y}_ttl", f"{y}_shw"), (len(ygrps), len(y_lrgst))))
-    df[[x, y]] = df[[x, y]].astype(str)
-
-    # final format
-    df = df.pivot_table(index=y, columns=x, values="cnt", fill_value=0, aggfunc="sum")
-    df = df.unstack().to_frame("cnt").reset_index()
+    xgrps, ygrps = df.groupby(x)["cnt"].sum(), df.groupby(y)["cnt"].sum()
+
+    if (
+        cfg.nested.enable
+        and cfg.stacked.enable
+        and cfg.heatmap.enable
+        and cfg.nested.ngroups == cfg.stacked.ngroups == cfg.heatmap.ngroups
+        and cfg.nested.nsubgroups == cfg.stacked.nsubgroups == cfg.heatmap.nsubgroups
+    ):
+        df, stats = parse_grps(df, cfg.nested.ngroups, cfg.nested.nsubgroups, x, y, xgrps, ygrps)
```

Comment:
Do not remove the comment. (`' parse the dataframe to consist of the nsubgroups largest y groups`)